src/grid/atomgrid.py

```diff
@@ -459,14 +427,18 @@ def _generate_atomic_grid(rad_grid, degs, rotate=False):
                 new_points = sphere_grid.points @ rot_mt
                 sphere_grid = AngularGrid(new_points, sphere_grid.weights)
             # construct atomic grid with each radial point and each spherical shell
-            points, weights = AtomicGrid._generate_sphere_grid(rad_grid[i], sphere_grid)
+            rad_order = 2
+            # compute points
+            points = sphere_grid.points * rgrid[i].points
+            # compute weights
+            weights = (
+                sphere_grid.weights * rgrid[i].weights * rgrid[i].points ** rad_order
```

Comment:
Is there any reason to define `rad_order = 2`? I think you can just replace it on line 435.